lib/api/src/instance.rs

```diff
@@ -96,6 +103,22 @@ impl Instance {
         })
     }
 
+    #[doc(hidden)]
+    #[deprecated = "this is an internal API and is subject to change"]
+    pub fn reinitialize(
+        &self,
+        store: &mut impl AsStoreMut,
+        module: &Module,
+    ) -> Result<(), InstantiationError> {
+        self._inner.reinitialize(store, module)
+    }
+
+    #[doc(hidden)]
+    #[deprecated = "this is an internal API and is subject to change"]
+    pub fn set_module(&mut self, module: &Module) {
+        self.module = module.clone();
+    }
```

Comment:
If we want to do this should be over a trait in sys

Reply:
This has been deleted